### PR TITLE
[SPARK-41397][CONNECT][PYTHON] Implement part of string/binary functions

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Implement the first half of string/binary functions. The rest of the string/binary functions will be implemented in a separate PR for easier review.

### Why are the changes needed?
For API coverage on Spark Connect.

### Does this PR introduce _any_ user-facing change?
Yes. New functions are available on Spark Connect.

### How was this patch tested?
Unit tests.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala [hunk 1]
```diff
@@ -44,16 +44,16 @@ import org.apache.spark.unsafe.types.{ByteArray, UTF8String}
 // This file defines expressions for string operations.
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-
```
Comment (zhengruifeng) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:47]: I think we need to revert the changes in this file?

Comment (xinrong-meng) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala]: Thanks!

**Comment by xinrong-meng** [comment]
@amaliujia @HyukjinKwon @grundprinzip Would you please review? Thanks!

**Review by amaliujia — APPROVED** [review]

File: python/pyspark/sql/tests/connect/test_connect_function.py [hunk 2]
```diff
@@ -410,6 +410,67 @@ def test_aggregation_functions(self):
...
+            SELECT * FROM VALUES
+            ('   ab   ', 'ab   ', NULL), ('   ab', NULL, 'ab')
+            AS tab(a, b, c)
+            """
+        # +--------+-----+----+
+        # |       a|    b|   c|
+        # +--------+-----+----+
+        # |   ab   |ab   |null|
+        # |      ab| null|  ab|
+        # +--------+-----+----+
```
[the remark this answers is not in the document]
Comment (xinrong-meng) [reply python/pyspark/sql/tests/connect/test_connect_function.py:427]: Thanks!

**Review by HyukjinKwon — APPROVED** [review]
LGTM but let me leave it to @zhengruifeng

**Review by zhengruifeng — APPROVED** [review]
LGTM

as to `decode`, I think we can deal with it in the newly added `transformUnregisteredFunction`

**Comment by zhengruifeng** [comment]
merged into master
